server/configuration-validation.js

```diff
@@ -1,125 +1,192 @@
+        typeof populateEntryRule !== 'string'
+      ) {
+        log.error(
+          `the "populateEntryRule" option of "${name}" should be an object/array/string`
+        )
+      } else if (populateEntryRule !== undefined) {
+        options.populateEntryRule = populateEntryRule
+      }
+
+      return this
+    },
+    validateNoInvalidKeys() {
+      // Keys that should not be present in the configuration
+      Object.keys(excedent).map(key => {
+        log.warn(`The attribute "${key}" of "${name}" is not a known option`)
+      })
+
+      return this
+    },
+    get() {
+      return options
+    },
   }
 }
 
-function validateCollectionConfiguration({ configuration, collection }) {
-  const validApiFields = [
-    'indexName',
-    'transformEntry',
-    'settings',
-    'filterEntry',
-    'populateEntryRule',
-  ]
+function PluginOptions(configuration) {
+  const log = strapi.log // has to be inside a scope
+  const { apiKey, host, ...collections } = configuration
+  const options = {}
 
-  if (configuration === undefined) {
-    return
-  }
+  return {
+    validateConfiguration() {
+      // Configuration must be an object
+      if (!isObject(configuration)) {
+        log.error(
+          'The `config` field in the Meilisearch plugin configuration must be of type object'
+        )
+      }
 
-  if (configuration !== undefined && !isObject(configuration)) {
-    strapi.log.error(`The collection "${collection}" should be of type object`)
-    return {}
-  }
+      return this
+    },
 
-  const { indexName } = configuration
-  if (
-    (indexName !== undefined && typeof indexName !== 'string') ||
-    indexName === ''
-  ) {
-    strapi.log.error(
-      `the "indexName" param of "${collection}" should be a non-empty string`
-    )
-    delete configuration.indexName
-  }
-  if (
-    configuration.transformEntry !== undefined &&
-    typeof configuration.transformEntry !== 'function'
-  ) {
-    strapi.log.error(
-      `the "transformEntry" param of "${collection}" should be a function`
-    )
-    delete configuration.transformEntry
-  }
+    validateApiKey() {
+      // apiKey is either undefined or a string
+      if (apiKey !== undefined && typeof apiKey !== 'string') {
+        log.error(
+          '`apiKey` should be a string in Meilisearch plugin configuration'
+        )
+      } else if (apiKey !== undefined) {
+        options.apiKey = apiKey
+      }
+      return this
+    },
 
-  if (
-    configuration.filterEntry !== undefined &&
-    typeof configuration.filterEntry !== 'function'
-  ) {
-    strapi.log.error(
-      `the "filterEntry" param of "${collection}" should be a function`
-    )
-    delete configuration.filterEntry
-  }
+    validateHost() {
+      // // apiKey is either undefined or a none empty string
+      if ((host !== undefined && typeof host !== 'string') || host === '') {
+        log.error(
+          '`host` should be a non-empty string in Meilisearch plugin configuration'
```

Comment:
Also sometimes you go with `The x option is missing` so I guess you should do `The apiKey option should be...`